pkg/image/prune/prune.go

```diff
@@ -455,18 +475,36 @@ func addPodSpecToGraph(g graph.Graph, spec *kapi.PodSpec, predecessor gonum.Node
 		}
 
 		if len(ref.ID) == 0 {
+			ref = ref.DockerClientDefaults()
+			if p.registryURL.Host != ref.RegistryURL().Host {
```

Comment:
This is very fragile IMHO. There are many possible registry hosts that can be used to fetch images from the integrated registry. Currently, this considers just the one determined by the algorithm or passed with `--registry-url`, which may be totally wrong.

Shall we collect all the possible registry hosts from all the images when we build the graph and match the url against the collected set?

Reply:
how would you build the set of possible registry hosts that represent the internal registry?

Reply:
Collecting all hosts from all the images is not enough. In general we can't know which DockerImageReference uses the internal registry. One can pull images through the router.

Even now I can use three names for the integrated registry: 172.30.1.1:5000, docker-registry.default.svc and docker-registry.default.svc.cluster.local. But only one can be specified in p.registryURL.

Reply:
right, that's why i'm asking how you could build the set of all possible hosts that represent the registry?  it doesn't seem possible (or at least reasonably possible) since anyone can create a service or route that points to it and then use that.

So i'm trying to understand what you're proposing doing when you say: 

> Shall we collect all the possible registry hosts from all the images when we build the graph and match the url against the collected set?

Reply:
Most of the images pushed to the integrated registry will have `$OPENSHIFT_DEFAULT_REGISTRY` host name in their `dockerImageReference` attribute. All of them will be managed (and having managed annotation). There may be even some older managed images having some obsolete registry address. These addresses are most probably being used inside a cluster in deployments, pods, daemonsets, etc.

For the external ones, it's much harder. All we have (sometimes) is the optional `--registry-url` flag.

What I had in mind is iterating over all managed images and get a set of internal addresses, add the `registry-url` and `docker-registry.default.svc.cluster.local` and compare image references against this set.